io/disk/iozone.py

```diff
@@ -0,0 +1,649 @@
+                    w_count = int(match.group(1))
+                    section = self.__get_section_name(match.group(2))
+
+                    # Output the appropriate keyval pair
+                    key_name = '%s-%d-kids' % (section, w_count)
+                    keylist[key_name] = match.group(3)
+                    continue
+
+                # Check for any other interesting lines
+                if '=' in line:
+                    # Is it something we recognize? First check for parent.
+                    match = parent_regexp.search(line)
+                    if match:
+                        # The section name and the worker count better match
+                        p_count = int(match.group(1))
+                        p_secnt = self.__get_section_name(match.group(2))
+                        if p_secnt != section or p_count != w_count:
+                            continue
+
+                        # Set the base name for the keyval
+                        basekey = 'parent'
+                    else:
+                        # Check for the various 'throughput' values
+                        if line[3:26] == ' throughput per thread ':
+                            basekey = line[0:3]
+                            match_x = KBsec_regexp
+                        else:
+                            # The only other thing we expect is 'Min xfer'
+                            if not line.startswith('Min xfer '):
+                                continue
+                            basekey = 'MinXfer'
+                            match_x = KBval_regexp
+
+                        match = match_x.search(line)
+                        if match:
+                            result = match.group(1)
+                            key_name = "%s-%d-%s" % (section, w_count, basekey)
+                            keylist[key_name] = result
+        output_path = os.path.join(self.outputdir, "perf.json")
+        json.dump(keylist, open(output_path, "w"), indent=1)
+
+    def test(self):
+
+        dir = self.params.get('dir', default=None)
+        args = self.params.get('args', default=None)
+
+        if not dir:
+            dir = self.base_dir
+        os.chdir(dir)
+
+        if not args:
+            args = '-a'
+
+        cmd = os.path.join(self.srcdir, 'src', 'current', 'iozone')
+        self.results = process.system_output('%s %s' % (cmd, args))
+        self.auto_mode = ("-a" in args)
+        iteration = 1
+        self.results_path = os.path.join(self.outputdir,
+                                         'raw_output_%s' % iteration)
+        self.analysisdir = os.path.join(self.outputdir,
+                                        'analysis_%s' % iteration)
+        f = open(self.results_path, 'w')
+        try:
+            f.write(self.results)
+        finally:
+            f.close()
+
+        self.generate_keyval()
+
+        if self.auto_mode:
+            a = IOzoneAnalyzer(list_files=[self.results_path],
```

Comment:
OK so the comparison is actually never used as you only support single `list_files` file. How about allowing `params.get(previous_results, None)` and if not None append it here so people might actually use it?

I tried it manually by providing previous data results and it seems to be working:

```
ANALYSIS of DRILLED DATA:

TABLE:  RECsize Difference between runs                 Results are % DIFF

RECORD    INIT    RE              RE    RANDOM  RANDOM  BACKWD   RECRE  STRIDE    F       FRE     F       FRE 
SIZE (KB) WRITE   WRITE   READ    READ    READ   WRITE  READ   WRITE    READ    WRITE   WRITE   READ    READ
--------------------------------------------------------------------------------------------------------------
4         +6.262  .       .       -8.366  -9.828  -10.40  +6.320  .       -6.743  -6.660  .       -12.33  .       
8         .       -11.74  .       -21.41  -9.530  .       -17.19  .       -20.49  .       -8.594  .       +8.815  
16        -8.796  .       +15.52  +5.163  +12.70  +6.207  +8.601  .       -16.18  -15.68  -11.26  .       +12.22  
32        .       .       -15.59  -18.29  .       -12.12  -18.49  -8.060  -20.69  -6.581  .       -9.863  -15.73  
64        .       .       .       -7.431  -9.286  .       -9.756  .       -10.61  .       .       .       -7.233  
128       .       .       .       .       -6.704  .       -11.20  .       .       .       -6.741  .       -12.61  
256       -8.730  .       .       .       .       .       +4.246  .       .       .       .       .       .       
512       .       +4.160  .       -9.914  -12.89  -6.136  .       -6.050  -8.972  -7.337  .       .       -10.27  
1024      .       .       .       .       .       .       -7.605  .       -12.47  .       .       .       -7.058  
2048      .       .       .       -8.330  -8.480  .       -6.871  -7.931  .       .       .       .       .       
4096      .       .       .       +7.829  +7.992  +9.787  +4.638  +13.42  .       .       .       .       .       
8192      .       .       .       .       .       +7.965  +5.080  +8.345  +5.943  .       .       -11.26  +7.264  
16384     -8.722  .       .       .       .       .       .       .       .       .       -8.733  .       .       
REGRESSIONS: 52 (28.57%)    Improvements: 21 (11.54%)


TABLE:  FILEsize Difference between runs               Results are % DIFF

RECORD    INIT    RE              RE    RANDOM  RANDOM BACKWD   RECRE  STRIDE    F       FRE     F       FRE 
SIZE (KB) WRITE   WRITE   READ    READ    READ   WRITE READ   WRITE    READ    WRITE   WRITE   READ    READ
--------------------------------------------------------------------------------------------------------------
64        .       -9.863  .       -16.26  -7.558  -12.22  +11.23  +12.43  -25.95  .       -13.02  .       +49.07  
128       +10.42  .       .       -13.60  -11.65  .       -28.90  -26.85  -11.80  -12.69  +4.483  -6.698  +29.96  
256       .       -13.05  .       .       .       -11.82  +18.94  +7.166  -9.394  .       +8.011  .       -7.661  
512       .       .       -8.629  .       +6.799  +4.632  -17.58  -6.751  -6.571  .       -15.49  .       -26.72  
1024      -8.580  .       .       -19.98  .       +4.641  -18.82  .       -19.62  .       .       .       -20.35  
2048      .       +7.071  -6.958  -17.84  -20.31  .       .       .       -9.327  .       .       -6.058  -16.23  
4096      .       .       -6.227  .       -15.18  -6.789  .       +4.406  -17.42  .       .       .       .       
8192      .       .       .       +4.258  +4.722  .       .       .       -10.76  .       .       .       .       
16384     .       .       -6.143  -6.089  .       .       .       .       .       .       .       .       .       
32768     +5.035  .       +4.427  +4.924  .       +6.517  .       .       .       .       .       .       .       
65536     .       .       .       .       .       +4.154  .       .       +7.563  .       .       .       .       
131072    .       .       .       .       .       +4.115  .       .       .       .       .       .       .       
262144    -9.357  .       -9.898  -6.150  .       .       -15.17  -7.130  -14.80  -6.117  -9.779  -10.18  .       
524288    -8.684  .       .       -7.075  -9.334  .       .       -7.676  .       .       .       -6.640  .       
REGRESSIONS: 55 (28.06%)    Improvements: 23 (11.73%)
```